lib/vmpooler/pool_manager.rb

```diff
@@ -735,21 +735,26 @@ def _check_pool(pool, provider)
       raise
     end
 
-    # create a provider object based on the providers/*.rb class that implements providers/base.rb
-    # provider_class: needs to match a provider class in providers/*.rb ie Vmpooler::PoolManager::Provider::X
-    # provider_name: should be a unique provider name
+    # Create a provider object, usually based on the providers/*.rb class, that implements providers/base.rb
+    # provider_class: Needs to match a class in the Vmpooler::PoolManager::Provider namespace. This is
+    #                 either as a gem in the LOADPATH or in providers/*.rb ie Vmpooler::PoolManager::Provider::X
+    # provider_name:  Should be a unique provider name
     #
     # returns an object Vmpooler::PoolManager::Provider::*
     # or raises an error if the class does not exist
     def create_provider_object(config, logger, metrics, provider_class, provider_name, options)
-      case provider_class
-      when 'vsphere'
-        Vmpooler::PoolManager::Provider::VSphere.new(config, logger, metrics, provider_name, options)
-      when 'dummy'
-        Vmpooler::PoolManager::Provider::Dummy.new(config, logger, metrics, provider_name, options)
-      else
-        raise("Provider '#{provider_class}' is unknown for pool with provider '#{provider_name}'")
-      end
+      # Can't use just the string name due to old jruby support.  Instead, need
+      # to use the long form
+      provider_klass = Object.const_get('Vmpooler').const_get('PoolManager').const_get('Provider')
```

Comment:
Since we know what class/module we're after here, do we really need the `const_get` for this part?

Or can this just be:
```ruby
provider_klass = Vmpooler::PoolManager::Provider
```